ecommerce_integrations/zenoti/doctype/zenoti_settings/zenoti_settings.py

```diff
@@ -0,0 +1,332 @@
+
+import frappe
+import requests
+from frappe import _
+from frappe.custom.doctype.custom_field.custom_field import create_custom_fields
+from frappe.model.document import Document
+from frappe.utils import add_to_date, cint, get_datetime
+
+from ecommerce_integrations.zenoti.purchase_transactions import process_purchase_orders
+from ecommerce_integrations.zenoti.sales_transactions import process_sales_invoices
+from ecommerce_integrations.zenoti.stock_reconciliation import process_stock_reconciliation
+from ecommerce_integrations.zenoti.utils import api_url, get_list_of_centers
+
+
+class ZenotiSettings(Document):
+	def validate(self):
+		url = api_url + "centers"
+		headers = {}
+		headers["Authorization"] = "apikey " + self.api_key
+		response = requests.request("GET", url=url, headers=headers)
+		if response.status_code != 200:
+			frappe.throw("Please verify the API Key")
+		check_for_opening_stock_reconciliation()
+		check_perpetual_inventory_disabled()
+		setup_custom_fields()
+		add_genders()
+		make_item_group()
+		make_item_tips()
+		self.add_gift_and_prepaid_card_as_payment_mode()
+
+	def add_gift_and_prepaid_card_as_payment_mode(self):
+		payment_mode = "Gift and Prepaid Card"
+		account = self.liability_income_account_for_gift_and_prepaid_cards
+		add_mode_of_payments(payment_mode, account, self.company)
+
+
+def add_mode_of_payments(payment_mode, account, company):
+	if not frappe.db.get_value("Mode of Payment", payment_mode):
+		doc = frappe.new_doc("Mode of Payment")
+		doc.mode_of_payment = payment_mode
+		doc.enabled = 1
+		doc.type = "General"
+		doc.set("accounts", [])
+		add_payment_mode_accounts(doc, account, company)
+		doc.insert()
+
+
+def add_payment_mode_accounts(doc, account, company):
+	account = account
+	payment_mode_accounts = {"company": company, "default_account": account}
+	doc.append("accounts", payment_mode_accounts)
+
+
+def check_for_opening_stock_reconciliation():
+	if not frappe.db.exists("Stock Reconciliation", {"purpose": "Opening Stock"}):
+		frappe.throw(
+			_(
+				'Please reconcile the stocks using Stock Reconciliation with purpose "Opening'
+				' Stock" before configuring this'
+			)
+		)
+
+
+def sync_invoices():
+	if frappe.db.get_single_value("Zenoti Settings", "enable_zenoti"):
+		check_perpetual_inventory_disabled()
+		last_sync = frappe.db.get_single_value("Zenoti Settings", "last_sync")
+		interval = frappe.db.get_single_value("Zenoti Settings", "sync_interval")
+		if last_sync and get_datetime() > get_datetime(add_to_date(last_sync, hours=cint(interval))):
+			error_logs = []
+			list_of_centers = get_list_of_centers()
+			if len(list_of_centers):
+				process_sales_invoices(list_of_centers, error_logs)
+				frappe.db.set_value("Zenoti Settings", "Zenoti Settings", "last_sync", get_datetime())
+				if len(error_logs):
+					make_error_log(error_logs)
+
+
+def sync_stocks():
+	if frappe.db.get_single_value("Zenoti Settings", "enable_zenoti"):
```

Comment:
```suggestion
	if cint(frappe.db.get_single_value("Zenoti Settings", "enable_zenoti")):
```

Cast this to integer, in some old versions of frappe this casting was improper which caused issues. Best not to rely on that and explicitly cast this.